### PR TITLE
[-Wsign-compare] warning fixes batch 13

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 1]
```diff
@@ -800,15 +800,15 @@ Status EagerRemoteExecute(EagerOperation* op, TensorHandle** retvals,
   DataTypeVector output_dtypes;
   TF_RETURN_IF_ERROR(GetOutputDTypes(op, &output_dtypes));
 
-  const size_t num_outputs = static_cast<int>(output_dtypes.size());
-  if (num_outputs != *num_retvals) {
+  const size_t num_outputs = output_dtypes.size();
+  if (static_cast<int>(num_outputs) != *num_retvals) {
```
Comment (mihaimaruseac) [inline tensorflow/core/common_runtime/eager/execute.cc:804]: Why not leave the case where it is? And only change type to `int` instead of `size_t`?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/shape_inference.cc [hunk 2]
```diff
@@ -3082,7 +3092,7 @@ ShapeInference::InferDegenerateDimensionBroadcastShape(HloOpcode operation,
   }
 
   // All arguments must be compatible with the program shape.
-  for (int i = 0; i < arg_shapes.size(); ++i) {
+  for (int i = 0, end = arg_shapes.size(); i < endwwwwwwwwwww; ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/shape_inference.cc:3095]: Typo?

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h [hunk 3]
```diff
@@ -149,7 +149,7 @@ class EagerServiceImpl {
     bool IsStale() {
       mutex_lock l(last_accessed_mu_);
       return (destroy_after_micros_ > 0 &&
-              (env_->env->NowMicros() - last_accessed_micros_) >
+              static_cast<int64>(env_->env->NowMicros() - last_accessed_micros_) >
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/eager/eager_service_impl.h:152]: Better extract this to a separate expression

File: tensorflow/core/grappler/optimizers/data/vectorization/parse_single_example_vectorizer.cc [hunk 4]
```diff
@@ -87,7 +87,7 @@ class ParseSingleExampleVectorizer : public Vectorizer {
     TF_RETURN_IF_ERROR(node_builder.Finalize(outer_scope, &new_node));
 
     // Add output mappings
-    for (size_t i = 0; i < node.num_outputs(); ++i) {
+    for (size_t i = 0; i < static_cast<size_t>(node.num_outputs()); ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/data/vectorization/parse_single_example_vectorizer.cc:90]: This is not consistent with the other rewrites of the loop header.

File: tensorflow/core/grappler/optimizers/data/vectorization_utils.cc [hunk 5]
```diff
@@ -251,7 +251,7 @@ Status Vectorization::AddConversionMapping(Node* op_node) {
 
   // The inputs for the node to be converted may already have been converted
   // themselves. For those that are not, we promote them to MapDefun outputs.
-  for (size_t i = 0; i < op_node->num_inputs(); ++i) {
+  for (size_t i = 0; i < static_cast<size_t>(op_node->num_inputs()); ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/data/vectorization_utils.cc:254]: This is not consistent with the other rewrites of the loop header.

File: tensorflow/core/grappler/optimizers/data/vectorization_utils.cc [hunk 6]
```diff
@@ -279,15 +279,15 @@ Status Vectorization::AddConversionMapping(Node* op_node) {
...
+  const int64 op_node_num_outputs = op_node->num_outputs();
+  if (op_node_num_outputs != outputs.size()) {
     return errors::Internal(
         "Number of vectorizer outputs does not match. Expected: ",
         op_node->num_outputs(), " Actual: ", outputs.size());
   }
 
   // Add output mappings.
-  for (size_t i = 0; i < op_node->num_outputs(); ++i) {
+  for (size_t i = 0; i < static_cast<size_t>(op_node->num_outputs()); ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/data/vectorization_utils.cc:290]: This is not consistent with the other rewrites of the loop header.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 7]
```diff
@@ -1567,7 +1567,7 @@ Status StridedSliceTransposer::PermuteMask(TransposeContext* context,
     return errors::InvalidArgument("invalid mask value: ", mask_i);
   }
   int result = 0;
-  for (int i = 0, iter_limit = context->src_to_dst.size(); i < iter_limit;
+  for (int i = 0, end = context->src_to_dst.size(); i < end;
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1570]: No longer a need to split in two lines

File: tensorflow/core/grappler/optimizers/scoped_allocator_optimizer.cc [hunk 8]
```diff
@@ -548,7 +548,7 @@ class UnaryElementwiseRewriter : public ScopedAllocatorOptimizer::Rewriter {
       std::vector<InputDesc> inputs_to_first;
       LOG_WARNING_AND_RETURN_IF_ERROR(GetDataInputs(
           graph, sa_opti->node_map(), nd.from_node_def, &inputs_to_first));
-      for (int i = 0, iter_limit = inputs_to_first.size(); i < iter_limit;
+      for (int i = 0, end = inputs_to_first.size(); i < end;
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/scoped_allocator_optimizer.cc:551]: no longer a need to split in two lines

File: tensorflow/core/profiler/utils/event_span.cc [hunk 9]
```diff
@@ -128,7 +128,7 @@ std::vector<EventTypeSpan> ToNonOverlappedEvents(
   if (event_boundaries.empty()) return result;
   result.reserve(event_boundaries.size());
   PriorityTracker priority_tracker;
-  for (int64 i = 0, iter_limit = (event_boundaries.size() - 1); i < iter_limit;
+  for (int64 i = 0, end = (event_boundaries.size() - 1); i < end;
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/event_span.cc:131]: No longer a need to have it split in two lines

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc [hunk 10]
```diff
@@ -45,7 +45,7 @@ ::tensorflow::Status DropFakeQuant::Run(Model* model, std::size_t op_index,
   }
 
   // Drop min/max inputs
-  for (int i = 1, iter_limit = fakequant_op->inputs.size(); i < iter_limit;
+  for (int i = 1, end = fakequant_op->inputs.size(); i < end;
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:48]: Let's merge the two lines

File: tensorflow/python/client/session_ref.cc [hunk 11]
```diff
@@ -146,7 +146,7 @@ class SessionLogger {
     // Build an index from fetch tensor name to first index in
     // output_tensor_names.
     std::unordered_map<string, int> output_name_to_offset;
-    for (int i = 0, iter_limit = output_tensor_names.size(); i < iter_limit;
+    for (int i = 0, end = output_tensor_names.size(); i < end;
```
Comment (mihaimaruseac) [inline tensorflow/python/client/session_ref.cc:149]: Let's merge the two lines

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc (hunk 10 above)
Comment (tg-at-google) [reply tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:48]: handled in batch 11.

File: tensorflow/python/client/session_ref.cc (hunk 11 above)
Comment (tg-at-google) [reply tensorflow/python/client/session_ref.cc:149]: handled in batch 11.

**Comment by tg-at-google** [comment]
@mihaimaruseac

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h [hunk 12]
```diff
@@ -148,9 +148,10 @@ class EagerServiceImpl {
 
     bool IsStale() {
       mutex_lock l(last_accessed_mu_);
+      const int64 destroy_after_micros_comparison_object = env_->env->NowMicros()
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/eager/eager_service_impl.h:151]: `time_passed`?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h (hunk 12 above)
Comment (tg-at-google) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: suggested change applied.

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h [hunk 13]
```diff
@@ -148,9 +148,9 @@ class EagerServiceImpl {
 
     bool IsStale() {
       mutex_lock l(last_accessed_mu_);
+      const int64 time_paased = env_->env->NowMicros() - last_accessed_micros_); 
       return (destroy_after_micros_ > 0 &&
-              (env_->env->NowMicros() - last_accessed_micros_) >
-                  destroy_after_micros_);
+              destroy_after_micros_comparison_object > destroy_after_micros_);
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/eager/eager_service_impl.h:153]: You should change name here too.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h (hunk 13 above)
Comment (tg-at-google) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: updated.

Comment (mihaimaruseac) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: I don't see the name changed.

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h [hunk 14]
```diff
@@ -148,9 +148,10 @@ class EagerServiceImpl {
 
     bool IsStale() {
       mutex_lock l(last_accessed_mu_);
+      const int64 time_passed = env_->env->NowMicros()
+                                - last_accessed_micros_);
       return (destroy_after_micros_ > 0 &&
-              (env_->env->NowMicros() - last_accessed_micros_) >
-                  destroy_after_micros_);
+              time_passed > destroy_after_micros_);
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/eager/eager_service_impl.h:154]: Can this go on the same line as above?

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h (hunk 14 above)
Comment (tg-at-google) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: updated.

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h [hunk 15]
```diff
@@ -148,9 +148,9 @@ class EagerServiceImpl {
 
     bool IsStale() {
       mutex_lock l(last_accessed_mu_);
-      return (destroy_after_micros_ > 0 &&
-              (env_->env->NowMicros() - last_accessed_micros_) >
-                  destroy_after_micros_);
+      const int64 time_passed = env_->env->NowMicros()
+                                - last_accessed_micros_);
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/eager/eager_service_impl.h:152]: This can also go on the same line, I think

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/distributed_runtime/eager/eager_service_impl.h (hunk 15 above)
Comment (tg-at-google) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: updated.

Comment (tg-at-google) [reply tensorflow/core/distributed_runtime/eager/eager_service_impl.h]: updated.

**Comment by mihaimaruseac** [comment]
Closing as it has been handled by new PRs.

**Pull request closed without merging**